volmdlr/faces.py

```diff
@@ -2856,33 +2836,111 @@ def point2d_with_dimension_to_3d(self, point2d, points_x, points_y, xmin, xmax,
             Y=0
         elif Y>1:
             Y=1
+        
+        return volmdlr.Point2D(X,Y) 
+        
+    
+    def point2d_with_dimension_to_3d(self, point2d, points_x, points_y, xmin, xmax, ymin, ymax):
+        '''
+        compute the point3d, on a Bspline surface, of a point2d define in the dimensioned frame
+        '''
+        
+        point2d_01 = self.point2d_with_dimension_to_parametric_frame(point2d, points_x, points_y, xmin, xmax, ymin, ymax)
+        
+        return self.point2d_to_3d(point2d_01)
+    
 
-        return self.point2d_to_3d(volmdlr.Point2D(X,Y))
-
-
-    def contour2d_with_dimension_to_3d(self, contour2d):
+    def contour2d_parametric_to_dimension(self, contour2d:volmdlr.wires.Contour2D, points_x, points_y):
+        ''' 
+        convert a contour2d from the parametric to the dimensioned frame
         '''
         
+        xmin, xmax, ymin, ymax = 0, 1, 0, 1
+        point2d_dim = []
+        for primitive in contour2d.primitives:
+            point2d_dim.append(self.point2d_parametric_to_dimension(primitive.start, points_x, points_y, xmin, xmax, ymin, ymax))
+            if type(primitive) == volmdlr.edges.Arc2D:
+                point2d_dim.append(self.point2d_parametric_to_dimension(primitive.interior, points_x, points_y, xmin, xmax, ymin, ymax))
+        
+        return volmdlr.wires.Contour2D.from_points(point2d_dim)
+        
+    
+    def contour3d_to_2d_with_dimension(self, contour3d:volmdlr.wires.Contour3D, points_x, points_y): 
         '''
+        compute the contou2d of a contour3d, on a Bspline surface, in the dimensioned frame  
+        '''
+        
+        contour2d_01 = self.contour3d_to_2d(contour3d)
+        
+        return self.contour2d_parametric_to_dimension(contour2d_01, points_x, points_y)
+                   
 
-        for cle in self._grids2d.keys():
+    def contour2d_with_dimension_to_parametric_frame(self, contour2d):
+        ''' 
+        convert a contour2d from the dimensioned to the parametric frame
+        '''
+        
+        for cle in self._grids2d.keys(): 
             [points_x, points_y, xmin, xmax, ymin, ymax] = cle
-
+        
+        contour2d = contour2d.order_contour()
```

Comment:
I needed to order contour's primitives because, the method is based on start points (_it was necessary, to avoid missing points_) 
ps. This part of code is no longer needed. I improved it in another branch!